rest_api/sawtooth_rest_api/route_handlers.py

```diff
@@ -86,40 +129,43 @@ def __init__(
             link: /batches or /batch_status link for submitted batches
 
         """
+        timer_ctx = self._post_batches_total_time.time()
```

Comment:
Should this be a python Context?  What happens if an exception is thrown before the later stop call?

Reply:
Within time(), a new TimerContext() is created and returned.  The Timer instance itself does not keep a reference to the TimerContext, so there is no resource leak as far as I can tell. If an exception is thrown, stop() will not be called, which means this is, in it's currently form, a measurement of happy-path time.

Reply:
If stop is not called, is a measurement not actually made?

Reply:
Correct, timer_ctx basically does nothing until stop() is called, other than record it's creation time.